File: app/src/main/java/com/bluelock/fbdownloader/di/DownloadAPIInterface.java
package com.bluelock.fbdownloader.di;

import com.bluelock.fbdownloader.models.FacebookReel;
import com.bluelock.fbdownloader.models.FacebookVideo;
import com.bluelock.fbdownloader.models.InstaVideo;
import com.bluelock.fbdownloader.models.LikeeVideo;
import com.bluelock.fbdownloader.models.MojVideo;
import com.bluelock.fbdownloader.models.SnapVideo;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface DownloadAPIInterface {
    @GET("/link.php")
    Call<FacebookVideo> getFacebookVideos(@Query("video") String videoUrl);

    @GET("/instagram.php")
    Call<InstaVideo> getInstaVideos(@Query("video") String videoUrl);

    @GET("/snap.php")
    Call<SnapVideo> getSnapVideos(@Query("video") String videoUrl);

    @GET("/instagram.php")
    Call<LikeeVideo> getLikeeVideos(@Query("video") String videoUrl);

    @GET("/instagram.php")
    Call<MojVideo> getMojVideos(@Query("video") String videoUrl);

    @GET("/instagram.php")
    Call<FacebookReel> getFacebookReels(@Query("video") String videoUrl);
}

File: app/src/main/java/com/bluelock/fb/di/DownloadAPIInterface.kt
package com.bluelock.fb.di

import com.bluelock.fb.models.FacebookReel
import com.bluelock.fb.models.FacebookVideo
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface DownloadAPIInterface {
    @GET("/link.php")
    fun getFacebookVideos(@Query("video") videoUrl: String?): Call<FacebookVideo?>?

    @GET("/instagram.php")
    fun getFacebookReels(@Query("video") videoUrl: String?): Call<FacebookReel?>?
}